### PR TITLE
Robustify bundling

### PR DESCRIPTION
**Pull request by maackle**
I ran into failed tests if I ran `npm install` in src/cli/js-tests-scaffold. The reason was that the file `./src/cli/js-tests-scaffold/node_modules/node-releases/data/processed/envs.json` was unable to be parsed into an Object by serde_json in `bundle_recurse`, because that file contained an Array at the top-level rather than an Object. It caused two tests to fail, one of which was `cli::test::tests::test_command_no_npm`

One easy fix for this was to add `node_modules/` to the `.gitignore` for js-tests-scaffold, but that seems too brittle, and I'm not sure how easy it would be to track this down in real life. 

## tl;dr

In this PR I attempt to ignore unparseable JSON files discovered through tree traversal so random .json files won't throw a wrench in the works.

### CODE REVIEW

**Comment by Connoropolous** [comment]
Good thing you found this edge case first :)

File: src/cli/new.rs [hunk 1]
```diff
@@ -88,7 +93,7 @@ pub mod tests {
         // this test is deterministic
         let dir = gen_dir();
         let dir_path_buf = &dir.path().to_path_buf();
-        setup_test_folder(dir_path_buf, &TEST_DIR_NAME);
+        setup_test_folder(dir_path_buf, &TEST_DIR_NAME).expect("Test folder not set up");
```
Comment (Connoropolous) [inline src/cli/new.rs]: I think this may cause a panclic, which I'm not sure is what we want

File: src/cli/package.rs [hunk 2]
```diff
@@ -95,7 +95,8 @@ impl Packager {
 
             let json_file = fs::read_to_string(json_file_path)?;
 
-            serde_json::from_str(&json_file)?
+            // if the json file does not contain an Object at the top level, we can't parse it
```
Comment (Connoropolous) [inline src/cli/package.rs]: What does unwrap_or_default do here?

Comment (maackle) [reply src/cli/package.rs]: @Connoropolous  If `Err`, uses the Default value for the type getting unwrapped to. So in this case the default is just `Object::new()`, or the empty Map.

Comment (Connoropolous) [reply src/cli/package.rs]: Cool

File: src/cli/new.rs (hunk 1 above)
Comment (Connoropolous) [reply src/cli/new.rs]: Why would it fail to set up the test folder? and what would we want it to do, if it does?

Comment (maackle) [reply src/cli/new.rs]: I added this only to squelch a warning about an unused Result. So, it's better to panic than nothing at all! This would be really weird if it failed.

Comment (sphinxc0re) [reply src/cli/new.rs]: This is a test. So a panic is okay.

Comment (Connoropolous) [reply src/cli/new.rs]: Right, haha. oops.
